src/components/views/dialogs/SetMxIdDialog.js

```diff
@@ -106,6 +106,15 @@ export default React.createClass({
     },
 
     _doUsernameCheck: function() {
+        // XXX: SPEC-1
+        // Check if username is valid
+        if (encodeURIComponent(this.state.username) !== this.state.username) {
```

Comment:
It's used as a local validity check, synapse 500's if it sees utf8 on this endpoint and leaves the user confused, I'm just copying the logic from the registration page for the same endpoint